service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderController.java

```diff
@@ -121,11 +122,23 @@ public AboutToStartOrSubmitCallbackResponse handlePopulateSelectorMidEvent(
         CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
 
         if (NO.getValue().equals(caseData.getOrderAppliesToAllChildren())) {
-            ChildSelector childSelector = ChildSelector.builder().build();
-            childSelector.generateChildCount(caseData.getAllChildren().size());
+            Optional<Integer> remainingChildIndex = childrenService.getRemainingChildIndex(caseData.getAllChildren());
+            if (remainingChildIndex.isPresent()) {
+                caseDetails.getData().put("remainingChildIndex", String.valueOf(remainingChildIndex.get()));
+                caseDetails.getData()
+                    .put("remainingChild", childrenService.getRemainingChildren(caseData.getAllChildren()));
+                caseDetails.getData().put("otherFinalOrderChildren",
+                    childrenService.getFinalOrderIssuedChildren(caseData.getAllChildren()));
+            } else {
+                caseDetails.getData()
+                    .put("children_label", childrenService.getChildrenLabel(caseData.getAllChildren()));
+                ChildSelector childSelector = ChildSelector.builder()
+                    .childCount(ChildSelector.generateChildCount(caseData.getAllChildren().size()))
+                    .hidden(ChildSelector.generatedHiddenList(caseData.getAllChildren()))
+                    .build();
+                caseDetails.getData().put("childSelector", childSelector);
+            }
```

Comment:
I'm thinking this is getting slightly too complex for a controller. Could it go into the children service?